src/sources/prometheus/mod.rs

```diff
@@ -47,7 +55,31 @@ impl crate::config::SourceConfig for PrometheusConfig {
             let base_uri = host.parse::<http::Uri>().context(super::UriParseError)?;
             urls.push(format!("{}metrics", base_uri));
         }
-        Ok(prometheus(urls, self.scrape_interval_secs, shutdown, out))
+
+        let mut headers = IndexMap::new();
+        if let Some(bearer_token_file) = &self.bearer_token_file {
+            let mut text = Vec::<u8>::new();
+            File::open(bearer_token_file)
+                .expect("unable to open file")
+                .read_to_end(&mut text)
+                .expect("unable to read file");
```

Comment:
It would be better to read file asynchronous. `fn build` should be changed to `fn build_async`. https://github.com/timberio/vector/blob/439a9cfdb48d976f19a0d2e5b84fc268752d93ba/src/config/mod.rs#L141

Reply:
@langecode if you will have any problem with changing file reading to non-blocking let us know, we will help :slightly_smiling_face: